pkg/core/resources/apis/mesh/dataplane_validator.go

```diff
@@ -14,8 +14,15 @@ func (d *DataplaneResource) Validate() error {
 
 func validateNetworking(networking *mesh_proto.Dataplane_Networking) validators.ValidationError {
 	var err validators.ValidationError
-	if len(networking.GetInbound()) == 0 {
-		err.AddViolation("inbound", "has to contain at least one inbound interface")
+	if len(networking.GetInbound()) == 0 && networking.Gateway == nil {
+		err.AddViolation("inbound", "has to contain at least one inbound interface or gateway")
+	}
+	if len(networking.GetInbound()) > 0 && networking.Gateway != nil {
+		err.AddViolation("inbound", "cannot be defined both with networking.gateway")
```

Comment:
I think, it would be more intuitive to report it as en error on `networking` rather than on `networking.inbound`